src/Compilers/Shared/CoreClrAnalyzerAssemblyLoader.cs

```diff
@@ -8,19 +8,29 @@ namespace Microsoft.CodeAnalysis
 {
     internal sealed class CoreClrAnalyzerAssemblyLoader : AnalyzerAssemblyLoader
     {
-        private readonly AssemblyLoadContext _loadContext;
+        private AssemblyLoadContext _loadContext;
 
         public CoreClrAnalyzerAssemblyLoader()
         {
-            _loadContext = AssemblyLoadContext.GetLoadContext(typeof(CoreClrAnalyzerAssemblyLoader).GetTypeInfo().Assembly);
+        }
 
-            _loadContext.Resolving += (context, name) =>
+        protected override Assembly LoadFromPathImpl(string fullPath)
+        {
+            // Initialize the _loadContext in lazy fashion
+            if (_loadContext == null)
             {
-                Debug.Assert(ReferenceEquals(context, _loadContext));
-                return Load(name.FullName);
-            };
-        }
+                AssemblyLoadContext loadContext = AssemblyLoadContext.GetLoadContext(typeof(CoreClrAnalyzerAssemblyLoader).GetTypeInfo().Assembly);
 
-        protected override Assembly LoadFromPathImpl(string fullPath) => _loadContext.LoadFromAssemblyPath(fullPath);
+                System.Threading.Interlocked.CompareExchange(ref _loadContext, loadContext, null);
+
+                _loadContext.Resolving += (context, name) =>
```

Comment:
The event handler should probably be added only if `CompareExchange(ref _loadContext, loadContext, null) == null)`.